lib/ansible/modules/cloud/vmware/vmware_content_deploy_template.py

```diff
@@ -0,0 +1,273 @@
+      - If set to C(present) and VM exists, no action is taken.
+      - If set to C(poweredon) and VM does not exists, then VM is created with powered on state.
+      - If set to C(poweredon) and VM exists, no action is taken.
+      type: str
+      required: False
+      default: 'present'
+      choices: [ 'present', 'poweredon' ]
+extends_documentation_fragment: vmware_rest_client.documentation
+'''
+
+EXAMPLES = r'''
+- name: Deploy Virtual Machine from template in content library
+  vmware_content_deploy_template:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    template: rhel_test_template
+    datastore: Shared_NFS_Volume
+    folder: vm
+    datacenter: Sample_DC_1
+    name: Sample_VM
+    resource_pool: test_rp
+    validate_certs: False
+    state: present
+  delegate_to: localhost
+
+- name: Deploy Virtual Machine from template in content library with PowerON State
+  vmware_content_deploy_template:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    template: rhel_test_template
+    datastore: Shared_NFS_Volume
+    folder: vm
+    datacenter: Sample_DC_1
+    name: Sample_VM
+    resource_pool: test_rp
+    validate_certs: False
+    state: poweredon
+  delegate_to: localhost
+'''
+
+RETURN = r'''
+vm_deploy_info:
+  description: Virtual machine deployment message and vm_id
+  returned: on success
+  type: dict
+  sample: {
+        "msg": "Deployed Virtual Machine 'Sample_VM'.",
+        "vm_id": "vm-1009"
+    }
+'''
+
+import uuid
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware_rest_client import VmwareRestClient
+from ansible.module_utils.vmware import PyVmomi
+from ansible.module_utils._text import to_native
+
+HAS_VAUTOMATION_PYTHON_SDK = False
+try:
+    from com.vmware.vcenter.vm_template_client import LibraryItems
+    HAS_VAUTOMATION_PYTHON_SDK = True
+except ImportError:
+    pass
+
+
+class VmwareContentDeployTemplate(VmwareRestClient):
+    def __init__(self, module):
+        """Constructor."""
+        super(VmwareContentDeployTemplate, self).__init__(module)
+        self.content_service = self.api_client
+        self.template_service = self.api_client.vcenter.vm_template.LibraryItems
+        self.template_name = self.params.get('template')
+        self.vm_name = self.params.get('name')
+        self.datacenter = self.params.get('datacenter')
+        self.datastore = self.params.get('datastore')
+        self.folder = self.params.get('folder')
+        self.resourcepool = self.params.get('resource_pool')
+        self.cluster = self.params.get('cluster')
+        self.host = self.params.get('host')
+
+    def deploy_vm_from_template(self, power_on=False):
+        # Find the datacenter by the given datacenter name
+        self.datacenter_id = self.get_datacenter_by_name(
```

Comment:
If we use `self.api_client` then this API will look like - 

```suggestion
        self.datacenter_id = self.get_datacenter_by_name(datacenter_name=self.datacenter)
```